### PR TITLE
fix getValues type

### LINKED ISSUE #1446: [typescript]type error, when set `control` props to `Controller` component;
**Describe the bug**
in the latest version; use with typescript, if declare form data type use `interface`,  will get a type error, when set `control` props to `Controller` component;

**To Reproduce**
see the example in Codesandbox as line 25

**Codesandbox link (Required)**
https://codesandbox.io/s/react-hook-form-useform-template-nsmib

**Expected behavior**
no type error

### PR DESCRIPTION
**Pull request by kotarella1110**
close #1446
[CodeSandbox](https://codesandbox.io/s/react-hook-form-useform-template-dw0z5)

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -283,9 +283,9 @@ export type Control<FormValues extends FieldValues = FieldValues> = {
       : LiteralToPrimitive<U>,
     shouldValidate?: boolean,
   ): void;
-  getValues(): IsFlatObject<FormValues> extends true
-    ? FormValues
-    : Record<string, unknown>;
+  getValues(): IsFlatObject<FormValues> extends false
+    ? Record<string, unknown>
+    : FormValues;
```
Comment (kotarella1110) [inline src/types.ts:288]: I just changed extends for Conditioal Types from `true` to `false`.

**Review by kotarella1110 — COMMENTED** [review]
This is a strange bug...
why...

**Comment by bluebill1049** [comment]
why is this impact `type` vs `interface`?

File: src/types.ts (hunk 1 above)
Comment (bluebill1049) [reply src/types.ts:288]: wow, that's so strange...

**Comment by kotarella1110** [comment]
@bluebill1049 Sorry for my late reply.
I don't know why this bug caused.
It seems to be related to the type error I commented here https://github.com/react-hook-form/react-hook-form/pull/1412#issuecomment-614740110.
The type definition is correct, but conditional types with generics (`FormValues`) seem to have cases where inference doesn't work properly.

**Comment by bluebill1049** [comment]
yea quite strange one would be good to know the reason (as learning), but not going to stress about it.

**Pull request merged**